### PR TITLE
refactor and test: bridging contracts

### PR DESCRIPTION
**Pull request by kupermind**
- Addressing internal audit;
- Adding tests

### CODE REVIEW

File: contracts/bridges/BridgeMessenger.sol [hunk 1]
```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {IBridgeErrors} from "../interfaces/IBridgeErrors.sol";
+
+/// @title BridgeMessenger - Abstract smart contract for the bridge communication data processing
+/// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
+/// @author Andrey Lebedev - <andrey.lebedev@valory.xyz>
+/// @author Mariapia Moscatiello - <mariapia.moscatiello@valory.xyz>
+abstract contract BridgeMessenger is IBridgeErrors {
```
Comment (kupermind) [inline contracts/bridges/BridgeMessenger.sol:10]: This just extracts the data processing that is the same everywhere.

File: contracts/bridges/OptimismMessenger.sol [hunk 2]
```diff
@@ -1,169 +1,82 @@
...
 pragma solidity ^0.8.23;
 
+import {BridgeMessenger} from "./BridgeMessenger.sol";
+
 /// @dev Interface to the CrossDomainMessenger (CDM) Contract Proxy.
 interface ICrossDomainMessenger {
     function xDomainMessageSender() external returns (address);
 }
 
-/// @dev Provided zero address.
```
Comment (kupermind) [inline contracts/bridges/OptimismMessenger.sol:9]: Extracted into its own `IBridgeErrors.sol` file.

File: contracts/bridges/OptimismMessenger.sol [hunk 3]
```diff
@@ -1,169 +1,82 @@
...
-        }
-
-        // Check for the correct data length
-        uint256 dataLength = data.length;
-        if (dataLength < DEFAULT_DATA_LENGTH) {
-            revert IncorrectDataLength(DEFAULT_DATA_LENGTH, data.length);
+            revert SourceGovernorOnly(bridgeGovernor, governor);
         }
 
-        // Unpack and process the data
```
Comment (kupermind) [inline contracts/bridges/OptimismMessenger.sol:125]: Extracted into `BridgeMessenger` contract

File: contracts/bridges/WormholeMessenger.sol [hunk 4]
```diff
@@ -1,85 +1,37 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.23;
 
-/// @dev Provided zero address.
```
Comment (kupermind) [inline contracts/bridges/WormholeMessenger.sol:4]: Extracted into `IBridgeErrors.sol`

File: contracts/bridges/WormholeMessenger.sol [hunk 5]
```diff
@@ -154,54 +100,8 @@ contract WormholeMessenger {
         }
         mapDeliveryHashes[deliveryHash] = true;
 
-        // Check for the correct data length
```
Comment (kupermind) [inline contracts/bridges/WormholeMessenger.sol:157]: Extracted into `BridgeMessenger` contract

File: scripts/deployment/bridges/wormhole/test/messenger_sepolia_celo_alfajores_governor.js [hunk 6]
```diff
@@ -102,9 +102,9 @@ async function main() {
     const balanceETHBefore = await celoAlfajoresProvider.getBalance(EOAceloAlfajores.address);
 
     // Build the final payload to be passed from the imaginary Timelock
-    const sendPayloadSelector = "0x8fecdd02";
+    const sendPayloadSelector = "0x4b5ca6f4";
     const timelockPayload = await wormholeRelayer.interface.encodeFunctionData(sendPayloadSelector, [targetChain,
-        wormholeMessengerAddress, data, 0, minGasLimit]);
+        wormholeMessengerAddress, data, 0, minGasLimit, targetChain, wormholeMessengerAddress]);
```
Comment (kupermind) [inline scripts/deployment/bridges/wormhole/test/messenger_sepolia_celo_alfajores_governor.js:107]: This function usage is better for wormhole as it allows to collect undelivered funds.

File: test/xbuOLAS.js [hunk 7]
```diff
@@ -1,41 +0,0 @@
-/*global describe, context, beforeEach, it*/
```
Comment (kupermind) [inline test/xbuOLAS.js:1]: This became obsolete now.

**Review by mariapiamo — APPROVED** [review]

**Pull request merged**